File: Ld2_181RDB178.py
import math
import random
import array

def manualaIavede (rows, col): 
    arr=[]
    for i in range (rows):
        a=[]
        for j in range(col):
            print ("Ievadiet masīva elementu ", i, j)
            a.append(ievadeFloat())
        arr.append(a)   
    return arr    
def randIevade(rows, col):
    arr=[]
    for i in range(rows):
        a=[]
        for j in range(col):
            a.append(random.uniform(-50,50))
        arr.append(a)
    return arr
def parveidot(arr, rows, col):
    swapi=0
    swapj=1
    if rows%2==0:
        for i in range(0,rows,2):
            for j in range(col):
                arr[swapi][j],arr[swapj][j]=arr[swapj][j],arr[swapi][j]
        swapi=swapi+2
        swapj=swapj+2
    else:
        for i in range(0,rows-1,2):
            for j in range(col):
                arr[swapi][j],arr[swapj][j]=arr[swapj][j],arr[swapi][j]
        swapi=swapi+2
        swapj=swapj+2
def ievade():
    while True:
        user_input=input()
        try:
            valueX=int(user_input)
            return valueX
            break
        except ValueError:
            print("Kļūda ievadē")
def ievadeFloat():
    while True:
        user_input=input()
        try:
            valueFloat=float(user_input)
            return valueFloat
            break
        except ValueError:
            print ("Kļūda ievadē")
def masivaIzvade(arr, rows,col):
    print ("Masīvs: ")
    for i in range(rows):
        for j in range(col):
            print("%7.2f" %arr[i][j], end="|")
        print() 
    return arr
#main part
print("Jolanta Bērziņa 181RDB178  8.grupa")
print ('Ievadiet matricas rindu skaitu: ')
rows=ievade()
print('Ievadiet kollonu skaitu: ')
col=ievade()
answer=input("Vai skaitļus ievadīt manuāli: [y/n]")
if (answer=="y" or answer=="Y"):
    arr=manualaIavede(rows, col)
else:
    arr=randIevade(rows,col)
masivaIzvade(arr, rows,col)
parveidot(arr,rows,col)
print("Masīvs pēc pārveidošanas: ")
masivaIzvade(arr, rows,col)

File: Ld2_181RDB178.cpp
#include <iostream>
#include <stdio.h>
#include <iomanip>
#include <stdlib.h>
#include<ctime>

using namespace std;
double userInputDouble(){
    bool ok = false; double x;
    while (!ok)
    {
            cin >> x;
        if (!cin.fail() && (cin.peek() == EOF || cin.peek() == '\n'))
        {
            ok = true; return x;
        }
        else
        {
            cin.clear();
            cin.ignore(256, '\n');
            cout << "Ievades kļūda. Atkārtoti ievadiet skaitli: "  <<endl;
        }
    }
    return -1;
}
void manualaIevade(int Rinda, int Kolonna, double **mas){
   double x;
   for ( int i=0; i<Rinda; i++)
    {
        for (int j=0; j<Kolonna; j++)
        {
            printf("Ievadiet masīva %d %d elementu: ",i,j);
            x = userInputDouble();
            mas[i][j]=x;
            //cin>>mas[i][j];
            cout << "\n"; 
        }
    } 
}
void randomIevade(int Rinda, int Kolonna, double **mas){
    for ( int i=0; i<Rinda; i++)
    {
        for (int j=0; j<Kolonna; j++)
        {
          mas[i][j]=(float)rand()*101-51; 
        }
    } 
}

void masivaIzvade (int Rinda, int Kolonna, double **mas){
    cout<<"\nMasivs:"<<endl;
        for (int i=0; i<Rinda; i++) {
            cout << "\n";
            for (int j=0; j<Kolonna; j++){
                printf("%7.2f\t", mas[i][j]);
            }
        }
}

void masivaParveide (int rinda, int kollona, double **mas){
    int swapi=0; int swapj=1; float temp;
    if (rinda%2==0){
        for(int i=0; i<rinda; i=i+2){
            for(int j=0; j<kollona; j=j+2){
                temp=mas[swapi][j];
                mas[swapi][j]=mas[swapj][j];
                mas[swapj][j]=temp;
            }
        swapi=swapi+2;
        swapj=swapj+2;
        }
    }
    else {
        for(int i=0; i<rinda-1; i=i+2){
            for(int j=0; j<kollona; j=j+2){
               temp=mas[swapi][j];
                mas[swapi][j]=mas[swapj][j];
                mas[swapj][j]=temp; 
            }
        swapi=swapi+2;
        swapj=swapj+2;
        }
    }
}

int userInputInteger(string msg){
    bool ok = false;
        int x;
        cout << msg << "\n";
    while (!ok)
    {
        cin >> x;
        if (!cin.fail() && (cin.peek() == EOF || cin.peek() == '\n'))
        {
            ok = true; return x;
        }
        else
        {
            cin.clear();
            cin.ignore(256, '\n');
            cout << "Ievades kļūda. Atkārtoti ievadiet skaitli: " << endl;
        }
    }
            return -1;
}

int main( int argc, char *argv[])
{
    cout<<"Jolanta Bērziņa 181RDB178 8. grupa\n";
    int  rows, col, i;
    rows=userInputInteger("Ievadiet rindu skaitu:");
    col=userInputInteger("Ievadiet kollonu skaitu:");
    string arguments;
    //masīva izmera inicializacija
    double **mas=new double*[rows];
    for (int i=0; i<rows; i++){
        mas[i]=new double[col];
    }
    arguments=argv[i];
    if (arguments == "-rand") {
        randomIevade(rows, col, mas);
        masivaIzvade(rows,col, mas);
    }
    else {
        manualaIevade(rows,col, mas);
        masivaIzvade(rows,col,mas);
    }
    cout<<"\n"<<endl;
    cout<<"Masīvs pēc pārveidošanas: "<<endl;
    masivaParveide(rows, col, mas);
    masivaIzvade(rows,col,mas);
    
    return 0;
}


